scripts/packages/VSCodeServer/src/eval.jl

```diff
@@ -19,63 +19,75 @@ function repl_runcode_request(conn, params::ReplRunCodeRequestParams)
+                    else
+                        # Indent by 7 so that it aligns with the julia> prompt
+                        print(' '^7)
+                    end
 
-        withpath(source_filename) do
-            res = try
-                ans = inlineeval(resolved_mod, source_code, code_line, code_column, source_filename)
-                @eval Main ans = $(QuoteNode(ans))
-            catch err
-                EvalError(err, catch_backtrace())
+                    println(line)
+                end
             end
 
-            if show_result
-                if res isa EvalError
-                    Base.display_error(stderr, res)
+            fix_displays()
+
+            withpath(source_filename) do
+                res = try
+                    ans = inlineeval(resolved_mod, source_code, code_line, code_column, source_filename)
+                    @eval Main ans = $(QuoteNode(ans))
+                catch err
+                    EvalError(err, catch_backtrace())
+                end
+
+                if show_result
+                    if res isa EvalError
+                        Base.display_error(stderr, res)
 
-                elseif res !== nothing && !ends_with_semicolon(source_code)
+                    elseif res !== nothing && !ends_with_semicolon(source_code)
+                        try
+                            Base.invokelatest(display, res)
+                        catch err
+                            Base.display_error(stderr, err, catch_backtrace())
+                        end
+                    end
+                else
                     try
-                        Base.invokelatest(display, res)
+                        Base.invokelatest(display, InlineDisplay(), res)
                     catch err
-                        Base.display_error(stderr, err, catch_backtrace())
-                    end
-                end
-            else
-                try
-                    Base.invokelatest(display, InlineDisplay(), res)
-                catch err
-                    if !(err isa MethodError)
-                        printstyled(stderr, "Display Error: ", color = Base.error_color(), bold = true)
-                        Base.display_error(stderr, err, catch_backtrace())
+                        if !(err isa MethodError)
+                            printstyled(stderr, "Display Error: ", color = Base.error_color(), bold = true)
+                            Base.display_error(stderr, err, catch_backtrace())
+                        end
                     end
                 end
-            end
 
-            rendered_result = safe_render(res)
+                rendered_result = safe_render(res)
+            end
         end
     end
     return rendered_result
 end
 
+function fix_displays()
+    for d in reverse(Base.Multimedia.displays)
+        if d isa InlineDisplay
+            popdisplay(InlineDisplay())
+        end
+    end
+    pushdisplay(InlineDisplay())
+end
```

Comment:
Not clear what this is? Shouldn't be part of this PR, right?

Reply:
Ah, yeah, that was an effort to fix our display logic that snaked (I dunno why I always try to type "snuck" here...) in here by accident.